cmd/kubeadm/app/cmd/reset.go

```diff
@@ -295,3 +311,17 @@ func resetConfigDir(configPathDir, pkiPathDir string) {
 		}
 	}
 }
+
+// isControlPlane Checks if is control plane by checking if they have the cfg and if the KubeAPIServer static pod exists
+func isControlPlane(cfg *kubeadmapi.InitConfiguration) bool {
+	if cfg == nil {
```

Comment:
This may be true on a broken control plane node. Hence I think, that the real test is the existence of the API Server pod spec and this should be removed.